internal/core/src/segcore/ConcurrentVector.h

```diff
@@ -164,17 +166,21 @@ class ConcurrentVectorImpl : public VectorBase {
                                    BinaryVector>>>>;
 
  public:
-    explicit ConcurrentVectorImpl(ssize_t elements_per_row,
-                                  int64_t size_per_chunk)
+    explicit ConcurrentVectorImpl(
+        ssize_t elements_per_row,
+        int64_t size_per_chunk,
+        storage::MmapChunkDescriptor mmap_descriptor = nullptr)
         : VectorBase(size_per_chunk),
           elements_per_row_(is_type_entire_row ? 1 : elements_per_row) {
+        chunks_ptr_ = SelectChunkVectorPtr<Type>(mmap_descriptor);
     }
 
     Span<TraitType>
     get_span(int64_t chunk_id) const {
-        auto& chunk = get_chunk(chunk_id);
+        auto chunk_data = chunks_ptr_->get_chunk_data(chunk_id);
+        auto chunk_size = chunks_ptr_->get_chunk_size(chunk_id);
```

Comment:
could there be a possiblity size and data or not consistent?
maybe we can get size and data at the same time?

Reply:
get_chunk_data get a const void*, the size is no change